### PR TITLE
Set up Github Actions to run specs and rubocop on every commit

### PR DESCRIPTION
**Pull request by lavaturtle**
What it says on the tin.

Also included:
- Update `.ruby-version` file to 2.7.3 to match our other repos
- Get specs and rubocop passing

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.6.6
+2.7.3
```
Comment (anero) [inline .ruby-version]: We bumped `agra` to `2.7.4` due to a security fix, should we use that version instead?

File: engaging-networks-rest.gemspec [hunk 2]
```diff
@@ -1,79 +1,73 @@
 # Generated by juwelier
 # DO NOT EDIT THIS FILE DIRECTLY
 # Instead, edit Juwelier::Tasks in Rakefile, and run 'rake gemspec'
-# frozen_string_literal: true
```
Comment (anero) [inline engaging-networks-rest.gemspec:4]: Should we keep this to avoid all those `freeze` calls?

**Review by anero — APPROVED** [review]
:+1:

File: engaging-networks-rest.gemspec (hunk 2 above)
Comment (lavaturtle) [reply engaging-networks-rest.gemspec:4]: This file is auto-generated by `bundle exec rake gemspec`; AFAIK we have no control over the output of that task.

**Pull request merged**